### PR TITLE
TSK-1665: primary object reference can be used as complex parameter in a task query.

### PR DESCRIPTION
**Pull request by ryzheboka**
https://sonarcloud.io/summary/new_code?branch=TSK-1665&id=ryzheboka_taskana

---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

**Review by holgerhagen — CHANGES_REQUESTED** [review]
Have you actually verified the results of your test? To be honest I'm not sure if the object reference query is working at all at the moment. Therefore I would like to make sure that we do not implement a test on invalid data.

There is nothing wrong in this PR. I just don't see the link to the test data you are using.

File: rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java:230]: Please use `UrlEncode.encode()`, so that the object is human readable :)

File: pom.xml
Comment (gitgoodjhe) [inline pom.xml:47]: Why do we have this dependency update here?

File: .github/workflows/continuous-integration.yml
Comment (gitgoodjhe) [inline .github/workflows/continuous-integration.yml:86]: Why did this change in this PR?

File: rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java
Comment (gitgoodjhe) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java:228]: This does not fullfill our coding guidelines. Reference here: https://taskana.atlassian.net/wiki/spaces/TAS/pages/2634055697/Testing+Guidelines

**Review by gitgoodjhe — CHANGES_REQUESTED** [review]

**Comment by gitgoodjhe** [comment]
> Please fix your commits. For some reason this PR contains 2 dependabot commits

see comments above :P

**Review by gitgoodjhe — APPROVED** [review]

**Review by mustaphazorgati — APPROVED** [review]

**Comment by mustaphazorgati** [comment]
@holgerhagen we still need your approval before merging this PR :)

**Review by holgerhagen — APPROVED** [review]

**Pull request merged**
